### ISSUE TITLE
Support for an optional Tab key 

### ISSUE THREAD

**Issue by agateblue** [sspanak/tt9#669]
(Sorry, I posted by mistake without filling in the description)

In the keyboard shortcuts, there are several available shortcuts such as "add a word", "edit text", etc.

I was wondering if it would be possible to add a shortcut for a TAB key (like the one that usually exist on physical keyboard). My use case is that when I'm writing in a textarea or multiline text field with TT9, I have no easy way to switch to the next field and my only option is to use the touch screen. A tab key would likely solve that in my case.

If a shotcut isn't possible, maybe there's another way? To be honest, just having the ability to tab (regardless of using a shortcut, symbols from 1 or 0, etc.) would be enough for me.

Let me know if you need any additional clarification and thank you for your work!

**Comment by sspanak** [comment]
Actually, there used to be a TAB key for a short period of time, but it wasn't so useful, so I removed it. Here are the problems I've encountered:

1. TT9 is designed to process key presses only when visible. This means, you can not use its TAB key to jump to/from buttons, dropdowns and other elements that do not allow typing. It is possible to make an exception and allow the potential TAB key to work even when TT9 is invisible, but I am not keen on doing ugly hacks in the code.
2. Android does not work like a computer. Many apps do not define the previous/next targets correctly. The most notable examples are Facebook Messenger and Google Chat. I've tried connecting a computer keyboard to my phone, but even when using the hardware TAB key, I was not able to select the "send" button after typing out a message. In some other app (I don't remember which, but I remember I was quite frustrated), on the login screen, I was able to toggle between the email and password fields, but I was not able to select the "login" button. And there are many more cases when TAB will not function as you expect.
3. In Android, the virtual keyboards are meant to be text generators, rather true hardware keyboard replacements. Android is built around touchscreens and in some cases, it makes it completely impossible to avoid touching the screen.

In summary, I can add a TAB key, but it will not work as you think. 

Now, I am not sure about your use cases, but as an alternative solution, I suggest that you check out the [KeyMapper app](https://docs.keymapper.club/). The setup is a bit more difficult, but you can configure it to "touch" the screen upon pressing a hardware key or a combination of keys. It is intended to help you interact with the apps, while TT9 is mostly focused on typing. For this reason, it may be more suitable for you.

**Comment by agateblue** [comment]
That makes sense thank you for the thorough explanation!
